tianshou/data/batch.py

```diff
@@ -728,10 +729,9 @@ def split(self, size: int, shuffle: bool = True,
             indices = np.random.permutation(length)
         else:
             indices = np.arange(length)
-        count = length // size + \
-            (length % size > 0 and not merge_last or length < size)
-        if merge_last and (length % size == 0 or count == 0):
-            merge_last = False
+        merge_last = merge_last and length % size > 0 and length >= size
+        count = length // size + (1 - merge_last) * \
+            (length % size > 0 or length < size)
```

Comment:
That's strange. I think ``length < size`` implies ``length % size > 0``.

Reply:
but removing length%size=0 cannot pass the unit test